lib/liquid/standardfilters.rb

```diff
@@ -341,9 +341,13 @@ def floor(input)
       raise Liquid::FloatDomainError, e.message
     end
 
-    def default(input, default_value = "".freeze)
-      is_blank = input.respond_to?(:empty?) ? input.empty? : !input
-      is_blank ? default_value : input
+    def default(input, default_value = ''.freeze)
+      return default_value unless input
+      if input.respond_to?(:empty?) && input.empty?
```

Comment:
Sorry I didn't comment this in the previous review, but these two lines may as well be `if !input || input.respond_to ...`